From 047b0dd49b2f47996445067652a725e3eef1ac1f Mon Sep 17 00:00:00 2001
From: ZenWalker <scow@riseup.net>
Date: Wed, 17 Mar 2021 11:38:30 +0100
Subject: [PATCH] Makefile: DISTRO Mobian instead postmarketOS

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3d8f39d..b42bae3 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 
 USERCHROME_FILES := $(sort $(wildcard src/userChrome/*.css))
 HOMEPAGE_FILES := head.html distro_links.html bottom.html
-DISTRO := postmarketOS
+DISTRO := Mobian
 DESTDIR :=
 FIREFOX_DIR := /usr/lib/firefox
 
-- 
2.30.1

